x-pack/plugins/index_lifecycle_management/public/store/middleware/set_selected_policy_from_selected_template.js

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  fetchedIndexTemplate,
+  fetchPolicies,
+  setSelectedPolicy,
+  setPhaseData
+} from '../actions';
+import { getSelectedNodeAttrs, getPhaseData } from '../selectors';
+import { PHASE_WARM, PHASE_NODE_ATTRS, PHASE_COLD } from '../constants';
+
+export const setSelectedPolicyFromSelectedTemplate = store => next => async action => {
```

Comment:
Do we know what the role of this code is, and why middleware was chosen? I'm kind of a middleware noob.

Reply:
Middleware observes state changes in redux store and allows for actions to be dispatched in response to those changes.  This allows for dependent logic to be decoupled from the places that trigger state changes.  Here the selected policy is set based on the selected template, but the select template logic doesn't have to know about that.